packages/language-service/src/typescript_host.ts

```diff
@@ -289,13 +291,14 @@ export class TypeScriptServiceHost implements LanguageServiceHost {
     // Remove file versions that are no longer in the file and invalidate them.
     const missing = Array.from(this.fileVersions.keys()).filter(f => !seen.has(f));
     missing.forEach(f => {
+      upToDate = false;
       this.fileVersions.delete(f);
       this.staticSymbolResolver.invalidateFile(f);
     });
 
     this.lastProgram = program;
 
-    return false;
+    return upToDate;
```

Comment:
Prefer

```suggestion
    return missing.length === 0;
```

Eliminating the need for `upToDate` variable

Reply:
I still need to have `hasInvalidatedFiles` or `upToDate` variable because of I need to track version changes in line 285.

Reply:
How about

```typescript
const upToDate = missing.length === 0 && version === lastVersion;
return upToDate;
```

this way it's clear both conditions we are checking.